traitlets/traitlets.py

```diff
@@ -418,6 +420,10 @@ def _dynamic_default_callable(self, obj):
         return getattr(self, 'make_dynamic_default', None)
 
     def instance_init(self, obj):
+        if not hasattr(obj, '_trait_values'):
```

Comment:
I think #78 essentially replicates installing descriptors in `__init__` without causing that to break.